encode.go

```diff
@@ -521,14 +521,10 @@ func writeMap(wr writer, m interface{}) error {
 	}
 
 	pairs := length * 2
-	if pairs > 255 {
-		return errorNew("map contains too many elements")
-	}
-
-	l := buf.Len() + 1 // +1 for pairs byte
+	l := buf.Len()
 	switch {
-	case l < 256:
-		_, err := wr.Write([]byte{byte(typeCodeMap8), byte(l)})
+	case l < math.MaxUint8:
```

Comment:
I think the previous 256 was correct since 255 is in the correct range for a `uint8`. For the same reason the below `case l < math.MaxUint32` is incorrect. The consistency makes sense, so perhaps `l <= math.MaxUint8` would be appropriate.

Reply:
You're right, `l` is missing size of encoding, changed it.